src/node_buffer.cc

```diff
@@ -219,7 +219,10 @@ MaybeLocal<Object> New(Isolate* isolate,
   CHECK(actual <= length);
 
   if (actual < length) {
-    data = static_cast<char*>(realloc(data, actual));
+    char* new_data = static_cast<char*>(realloc(data, actual));
```

Comment:
We should rethink this in the future when switching over to `MaybeLocal<>`. Before recently allocating too much memory would have aborted, but now it will simply throw. But yes, this approach is incorrect.